pricing-service/src/main/java/fr/xebia/xke/micronaut/pricing/domain/Discount.java

```diff
@@ -0,0 +1,30 @@
+package fr.xebia.xke.micronaut.pricing.domain;
+
+import com.google.common.collect.Range;
+import lombok.Value;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+@Value
+public final class Discount {
+
+    private static final Range<Integer> ACCEPTABLE_RANGE = Range.closed(0, 100);
+    private static final Discount NO_DISCOUNT = percentage(0);
+
+    int percentage;
+
+    private Discount(final int percentage){
+        checkArgument(ACCEPTABLE_RANGE.contains(percentage),
+                "Invalid value: %s (expected: value in %s)", percentage, ACCEPTABLE_RANGE);
```

Comment:
hm, donc là il va envoyer "Invalid value: 101 (expected: value in 0, 1, 2, 3, 4, 5, 6...)" ?

Reply:
nope, la méthode toString de Range va nous donner quelque chose comme "expected: value in [0,100]"